examples/camera-app/linux/src/clusters/camera-avsettingsuserlevel-mgmt/camera-avsettingsuserlevel-manager.cpp

```diff
@@ -217,11 +217,13 @@ Status CameraAVSettingsUserLevelManager::DPTZRelativeMove(uint16_t aVideoStreamI
                 {
                     // if the delta would move us out of the cartesian plane of the sensor, limit to the left hand edge
                     //
-                    uint16_t x1Movement = ((deltaX < 0) && (abs(deltaX) > viewport.x1)) ? -viewport.x1 : deltaX;
-                    viewport.x1         = static_cast<uint16_t>(viewport.x1 + x1Movement);
+                    uint16_t x1Movement =
+                        ((deltaX < 0) && (abs(deltaX) > viewport.x1)) ? -viewport.x1 : static_cast<uint16_t>(deltaX);
```

Comment:
As the movement can be negative (i.e. panning left, or tilting up) this should probably be as below, although the above worked when I unit tested it with deltaX = -30.   Same applies for x2, y1, and y2 movement.
int16_t x1Movement = ((deltaX < 0) && (abs(deltaX) > viewport.x1)) ? static_cast<int16_t>(-viewport.x1) : deltaX;

Reply:
Updated. having said that, I believe this seems complex. Is it that we want `x1 += deltaX` except we want to clamp at 0? Maybe just doing int32_t would allow us to do:

```cpp
   int32_t newX1 = static_cast<int32_t>(viewport.x1) + deltaX;
   viewport.x1 = std::static_cast<uint16_t>(std::clamp<int32_t>(newX1, 0, 0xFFFF));
```

And maybe create a helper for that since it seems we use it 2x times.